core/cloudflow-spark/src/main/scala/cloudflow/spark/SparkStreamlet.scala

```diff
@@ -156,15 +156,22 @@ trait SparkStreamlet extends Streamlet[SparkStreamletContext] with Serializable
     context.streamletConfig.getString(configKey)
   }
 
-  private def makeSparkConfig(): SparkConf = {
-    val conf = new SparkConf()
+  private def makeSparkConfig(config: Config): SparkConf = {
+    var conf = new SparkConf()
     val master = conf.getOption("spark.master").getOrElse("local[2]")
-    conf.setMaster(master)
+    conf = conf.setMaster(master)
       // arbitrary number - default is 200 which is quite large for our sample apps
       // Needs to take a value passed through configuration, in case the user would like to override this.
       .set("spark.sql.shuffle.partitions", "20")
       .set("spark.sql.codegen.wholeStage", "false")
       .set("spark.shuffle.compress", "false")
+    import scala.collection.JavaConverters._
+    Try { config.getConfig("spark") }.toOption.foreach { cfg ⇒
```

Comment:
This behavior should be documented since Spark has a specific model for configuration order and precedence is important for the effect of a config property since there are some different ways to pass it eg. cli, env vars or at spark session creation time, as in this case. Is this config mapped from the Spark operator CR somehow or is it a custom external configmap which bypasses spark-submit?

Reply:
Yes this must still be documented. This is passed through via kubectl cloudflow, to what you set there, so that you can configure spark. Right now you cannot configure anything AFAIK. It is not mapped 'from' the CR (I don't know what you exactly mean with that, since we create the CR in the operator?). The other items (env var, cli <- I'm assuming some spark cli) are not available when running spark in cloudflow, right? So this just configures the spark session. Is there a problem with doing it that way? /cc @maasg

Reply:
Configuration options are part of the CR definition at the Spark Operator side. If you want to udpate the a config options eg. add a config: https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/blob/master/docs/api-docs.md 
you could use:
```
sparkConfmap[string]string | (Optional)SparkConf carries user-specified Spark configuration properties as they would use the “–conf” option in spark-submit.
```
If you do the update app should be re-launched with the new updates.